### ISSUE TITLE
Windows 10: Error: The system cannot find the path specified.

### ISSUE THREAD

**Issue by PatrikElfstrom** [imagemin/imagemin#199]
My friend gets this error when running gulp-imagemin on all jpg files.
This was called from a clean file with the same code as in the example.

I tried running imagemin without gulp and imagemin-cli and those seem to work.

```
events.js:160
     throw er; // Unhandled 'error' event
     ^
Error: The system cannot find the path specified.

   at notFoundError (C:\Projects\arbete\Collector\CollectorCorporate\Source\Collector.Corporate.Site\node_modules\cross-spawn-async\lib\enoent.js:8:11)
   at verifyENOENT (C:\Projects\arbete\Collector\CollectorCorporate\Source\Collector.Corporate.Site\node_modules\cross-spawn-async\lib\enoent.js:43:16)
   at ChildProcess.cp.emit (C:\Projects\arbete\Collector\CollectorCorporate\Source\Collector.Corporate.Site\node_modules\cross-spawn-async\lib\enoent.js:30:19)
   at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
```

**Comment by kevva** [comment]
Seems like one of the binaries is missing (probably jpegtran or mozjpeg). Did he get any errors upon installing `gulp-imagemin`?

**Comment by PatrikElfstrom** [comment]
No I don't think so. I believe all test were green. We reinstalled a bunch
of times but I'll have to confirm with him tomorrow.

**Comment by PatrikElfstrom** [comment]
We did not get any errors while installing.
It seems like there was permission issues that caused the error. 
I think the culprit is visual studio. We restarted the computer and tested before starting visual studio and it worked. But it did also work after we started visual studio so I am not sure that is the problem but it seems to be the most likely.
So it works not but I will report back if the error reoccurs and if I find the cause.

**Comment by snowcxt** [comment]
I got similar error on win 8.1. I restarted the computer. But it didn't work

```
[16:08:24] Error in plugin 'gulp-imagemin'
Message:
    The system cannot find the path specified.

Details:
    errno: ENOENT
    code: ENOENT
    syscall: spawn undefined
    cmd: C:\WINDOWS\system32\cmd.exe /s /c ""D:\Thomas\GitHub\MissDolphin-Email\
node_modules\optipng-bin\vendor\optipng.exe" "-strip" "all" "-clobber" "-fix" "-
o" "3" "-out" "C:\Users\tcui\AppData\Local\Temp\6c63ae13-f73b-42c6-9164-f710d244
c6b4" "C:\Users\tcui\AppData\Local\Temp\cfa05698-09e5-4312-811b-b9d18de074c6""
    stdout:
    stderr: The system cannot find the path specified.

    fileName: D:\Thomas\GitHub\MissDolphin-Email\src\assets\img\250x250.png
    domainEmitter: [object Object]
    domain: [object Object]
    domainThrown: false
```

**Comment by PatrikElfstrom** [comment]
@snowcxt Do you use visual studio in the same directory?

**Comment by snowcxt** [comment]
No

**Comment by crick64** [comment]
I fixed the relative issues by install npm inside the gulp-imagemin directory:

```
cd path\to\the\project\node_modules\gulp-imagemin
npm i
```

Then retry. Good luck ;)

**Comment by aikar** [comment]
For anyone who runs into this ticket trying to debug this error on why `npm run <script>` doesn't work on Windows for things in node_modules/.bin/ such as gulp, it relates to the fact that npm is using cmd.exe instead of bash for the child.

You can fix this with NPM 5.1.0+ using `npm config set script-shell bash`

**Comment by Dan503** [comment]
I was able to finally get it working by installing various things directly by using this command:

``````````````
npm install --save-dev gulp-imagemin imagemin-jpegtran imagemin-pngquant pngquant-bin
``````````````

**Comment by akumarcomviva** [comment]
I was also struggling with this error a lot. I am working behing corporate proxy and firewalls. Please go through following steps. 
1. Download the windows build tools and go through the steps mentioned in the below link
https://spin.atomicobject.com/2019/03/27/node-gyp-windows/
2. To check if the above step worked try "npm install kerberos".  If you are able to install kereberos without any errors then step 1 is succesfull.
3 Put the npm config to set the http and https proxy.
4 set the environment variable  NODE_TLS_REJECT_UNAUTHORIZED to 0
5. Delete the node_modules directory and build or npm install again.

Hopefully once you take all the above stpes your porblem should be solved.

**Comment by derick-montague** [comment]
> Seems like one of the binaries is missing (probably jpegtran or mozjpeg). Did he get any errors upon installing gulp-imagemin?

> You can fix this with NPM 5.1.0+ using `npm config set script-shell bash`

You will want to reinstall all your packages after running this. I didn't see any errors, but not all the packages were being installed.

**Comment by EmanueleCiriachi** [comment]
I have the same problem - and so far, no solution.
